### DISCUSSION TITLE
Epson LX-350 / LQ-350 (or other USB Dot-Matrix printers)

### DISCUSSION THREAD

**Discussion by pi43r** [OpenPrinting/cups#410] [Q&A] [answered]
Hi,

I was wondering if anyone tried the LX-350 or LQ-350 USB dot matrix printer on linux already?
I could not find any drivers on openprinting except for the [LX-300+](https://www.openprinting.org/printer/Epson/Epson-LX-300plus). There are also no drivers on the official page.

I have to buy it for a customer and will set it up remotely, so I need to make sure that it works.
It will mainly print pdfs on endless paper.

**Comment by zdohnal** [comment, marked as answer]
Hi @pi43r ,

I haven't used a dot matrix printer on Linux yet, but if I check models which are provided by CUPS, I see a model for ```Epson 9-Pin Series``` printers in ```Epson``` manufacturer group in CUPS Web UI - maybe it will work for you, since the model LX-350 is 9-Pin. And ```Epson 24-Pin Series``` is there as well, for LQ-350.

**Comment by pi43r** [comment]
Hi zdohnai,

Thank you for your reply!
I was able to find a used LX-350 in my area and as you mentioned the `9-Pin Series` drivers work on that one. So I ordered the LQ-350 and with this one I didn't even have to install it manually. It was found automatically and added to the printer list.

**Comment by jsmeix** [comment]
Only FYI as addendum my general thoughts about
using dot matrix printers:

I think printing on a dot matrix printer by using
an usual CUPS printer driver does not make much sense
because "using a printer driver" means to use the usual
printing filtering system plus printer driver software
to produce the final printer-specific printing data.

But the usual printing filtering system processing
is based on PostScript and PDF data and usual printer
driver software produces printer-specific PostScript or PDF
or printer-specific raster data (e.g. PCL or ESC/P raster data).

Usually the raster graphics mode on a dot matrix printer
is far too slow with insufficient "dotted" printout quality
to be useful in practice for printing nowadays documents
like web pages or office documents or photos
(in high resolution color).

I think nowadays dot matrix printers are only used to
print plain text characters which are sent directly
(i.e. without any filtering or driver in between)
to the printer device.

E.g. one has files that contain plain text characters
in the right encoding for a specific dot matrix printer
e.g. plain ASCII or another 8-bit single-byte encoding
like ISO-8859-1 or the IBM PC character encoding
but usually no multibyte encoding like UTF-8.

When one has files that contain characters in the right
encoding for a dot matrix printer or when one has an
application that outputs characters in the right encoding
for a dot matrix printer, then one can and should
send that "as is" to the dot matrix printer device
to get printouts with sufficient speed and quality
via the dot matrix printer's built-in character set.

**Reply by kepstin** [reply]
This is definitely true for Epson 9-pin dot matrix printers, which have a quite large dot size (the pins are 1/72" center-to-center vertically).

Specifically, you really want to make use of the printer's built-in fonts and font features (NLQ, multi-strike, underline, italic, box drawing glyphs, etc.) whenever possible in order to get high quality printing using fonts specifically laid to match the printer's capabilities (and to use printer features that aren't accessible in graphics mode, like vertical half-step double-strike), and that's not something which CUPS and modern apps are designed to deal with.

Historically, these printers weren't really used with "drivers", and instead applications would directly output the printer command codes (sometimes allowing you to specify custom command codes to use printers or printer features the app didn't know about). Typical DOS word processors used fixed with 80-column text modes that let you see how text would be laid out on a page at pica pitch, and the app would translate style options set on text directly to the matching features in the printer command language when printing.

When using scalable fonts in graphics mode on a 9-pin printer, the printing will be slower and the text will generally look worse than the printer's native text unless the font is well hinted and the rasterizer takes full advantage of that.

For Epson 24-pin printers… the smaller and tighter-spaced dots (pins are set in two banks of 12 pins, each bank has pins set 1/90" center-to-center vertically, and one bank is offset vertically by 1/180" from the other) are fine enough to do … ok renditions of scalable fonts. It's not as important to make use of the printer's built-in text features, except for the special case of draft or high-speed draft text printing. For general printing, the way the built-in fonts are handled doesn't give any special advantage over using graphics mode aside from reducing the amount of data transfer required.
